core/src/main/java/org/infinispan/container/DefaultDataContainer.java

```diff
@@ -356,20 +350,55 @@ public void onEntryRemoved(Entry<K, InternalCacheEntry<K, V>> entry) {
       }
    }
 
-   public static class EntryIterator<K, V> implements Iterator<InternalCacheEntry<K, V>> {
+   public class EntryIterator implements Iterator<InternalCacheEntry<K, V>> {
 
       private final Iterator<InternalCacheEntry<K, V>> it;
 
+      private InternalCacheEntry<K, V> next;
+
       EntryIterator(Iterator<InternalCacheEntry<K, V>> it){this.it=it;}
 
+      private InternalCacheEntry<K, V> getNext() {
+         while (it.hasNext()) {
+            InternalCacheEntry<K, V> entry = it.next();
+            long now;
+            if (entry.canExpire() && entry.isExpired((now = timeService.wallClockTime()))) {
```

Comment:
Sure I can change the wallClockTime.  Regular get does do removal which is why I added it here at the time.  Actually I am changing expiration removal in [1] as it will need new handling anyways.

[1] https://issues.jboss.org/browse/ISPN-694